<launch>

    <arg name="datafile" default="terrain.png"/>
    <arg name="max_height" default="1.0"/>

    <!-- Load the image. -->
    <node pkg="grid_map_demos" type="image_publisher.py" name="image_publisher" output="screen">
        <param name="image_path" value="$(find convex_plane_decomposition_ros)/data/$(arg datafile)"/>
        <param name="topic" value="/image"/>
    </node>

    <!-- Launch the grid map image to grid map demo node. -->
    <node pkg="grid_map_demos" type="image_to_gridmap_demo" name="image_to_gridmap_demo" output="screen">
        <param name="image_topic" value="/image"/>
        <param name="min_height" value="0.0"/>
        <param name="max_height" value="$(arg max_height)"/>
        <param name="resolution" value="0.04"/>
    </node>

    <node pkg="convex_plane_decomposition_ros" type="convex_plane_decomposition_ros_add_noise"
          name="convex_plane_decomposition_ros_add_noise"
          output="screen" launch-prefix="">
        <param name="noiseGauss" value="0.01"/>
        <param name="noiseUniform" value="0.01"/>
        <param name="outlier_percentage" value="5.0"/>
        <param name="blur" value="false"/>
        <param name="frequency" value="30.0"/>
        <param name="elevation_topic_in" value="/image_to_gridmap_demo/grid_map"/>
        <param name="elevation_topic_out" value="/elevation_mapping/elevation_map_raw"/>
        <param name="height_layer" value="elevation"/>
        <param name="imageName" value="$(find convex_plane_decomposition_ros)/data/elevationMap"/>
    </node>

    <!-- Static transform map to odom. x y z yaw pitch roll frame_id child_frame_id -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="map2odom_broadcaster"
          args="0.0 0.0 0.0 0.0 0.0 0.0 map odom"/>

    <!-- Launch the plane decomposition node. -->
    <include file="$(find convex_plane_decomposition_ros)/launch/convex_plane_decomposition.launch">
        <arg name="node_parameter_file" value="$(find convex_plane_decomposition_ros)/config/demo_node.yaml"/>
    </include>

    <!-- Launch the approximation demo. -->
    <node pkg="convex_plane_decomposition_ros" type="convex_plane_decomposition_ros_approximation_demo_node"
          name="convex_plane_decomposition_ros_approximation_demo_node"
          output="screen" launch-prefix=""/>

    <!-- Launch rviz. -->
    <node type="rviz" name="rviz" respawn="true" pkg="rviz"
          args="-d $(find convex_plane_decomposition_ros)/rviz/config_demo.rviz"/>

</launch>
